eng/common/pipelines/templates/steps/generate-sdk.yml

```diff
@@ -0,0 +1,172 @@
+  - name: SpecRepoName
+    type: string
+  - name: SdkRepoOwner
+    type: string
+  - name: SdkRepoName
+    type: string
+  - name: IntegrationSdkRepoOwner
+    type: string
+    default: ''
+  - name: IntegrationSdkRepoName
+    type: string
+    default: ''
+  - name: SpecRepoCommit
+    type: string
+    default: 'HEAD'
+  - name: SdkRepoCommit
+    type: string
+    default: 'HEAD'
+  - name: ConfigType
+    type: string
+    values:
+      - 'TypeSpec'
+      - 'OpenAPI'
+    default: 'TypeSpec'
+    displayName: 'API specification type'
+  - name: ConfigPath
+    type: string
+    default: 'specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml'
+    displayName: 'Path to API specification file (TypeSpec/OpenAPI)'
+  - name: SkipPullRequestCreation
+    type: boolean
+    default: false
+
+steps:
+- pwsh: |
+    $specRepoDirectory = "$(Pipeline.Workspace)/${{ parameters.SpecRepoName }}"
+    Write-Host "##vso[task.setvariable variable=SpecRepoDirectory]$specRepoDirectory"
+    Write-Host "SpecRepoDirectory variable set to: $specRepoDirectory"
+
+    $sdkRepoDirectory = "$(Pipeline.Workspace)/${{ parameters.SdkRepoName }}"
+    Write-Host "##vso[task.setvariable variable=SdkRepoDirectory]$sdkRepoDirectory"
+    Write-Host "SdkRepoDirectory variable set to: $sdkRepoDirectory"
+
+    $sdkRepoNameForSpecGen = "${{ parameters.SdkRepoName }}"
+    if ($sdkRepoNameForSpecGen.EndsWith('-for-net')) {
+      $sdkRepoNameForSpecGen = $sdkRepoNameForSpecGen + "-track2"
+    }
+    Write-Host "##vso[task.setvariable variable=SdkRepoNameForSpecGen]$sdkRepoNameForSpecGen"
+    Write-Host "SdkRepoNameForSpecGen variable set to: $sdkRepoNameForSpecGen"
+
+    # create variables to be used to create sdk pull request
+    $sdkRepoOwnerForPullRequest = '${{ parameters.SdkRepoOwner }}'
+    $sdkRepoNameForPullRequest = '${{ parameters.SdkRepoName }}'
+
+    if ((-not [string]::IsNullOrEmpty('${{ parameters.IntegrationSdkRepoOwner }}')) -and (-not [string]::IsNullOrEmpty('${{ parameters.IntegrationSdkRepoName }}'))) {
+      $sdkRepoOwnerForPullRequest = '${{ parameters.IntegrationSdkRepoOwner }}'
+      $sdkRepoNameForPullRequest = '${{ parameters.IntegrationSdkRepoName }}'
+    }
+    Write-Host "##vso[task.setvariable variable=SdkRepoOwnerForPullRequest]$sdkRepoOwnerForPullRequest"
+    Write-Host "SdkRepoOwnerForPullRequest variable set to: $sdkRepoOwnerForPullRequest"
+    Write-Host "##vso[task.setvariable variable=SdkRepoNameForPullRequest]$sdkRepoNameForPullRequest"
+    Write-Host "SdkRepoNameForPullRequest variable set to: $sdkRepoNameForPullRequest"
+  displayName: "Create Run Time Variables"
+
+- template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+  parameters:
+    Paths:
+    - '/*'
+    - '!sdk/**/test-recordings/*'
+    - '!sdk/**/recordings/*'
+    - '!sdk/**/SessionRecords/*'
+    - '!sdk/**/session-records/*'
+    Repositories:
+    - Name: ${{ parameters.SdkRepoOwner }}/${{ parameters.SdkRepoName }}
```

Comment:
What is the reason to move the spec checkout to the containing yml?

Reply:
Updated code per the discussion.